lib/bloc/lnurl/lnurl_bloc.dart

```diff
@@ -17,6 +17,12 @@ class LNUrlBloc with AsyncActionsHandler {
 
   StreamController _lnUrlStreamController;
 
+  final _newPayRequestController = StreamController<PayFetchResponse>();
```

Comment:
Why do we need this stream controller and its corresponding stream?
As far as I understand the fetch action is handled by the _listenLNUrl_ function that returns a stream directly to the _lnurl_handler.dart_.

Reply:
> Why do we need this stream controller and its corresponding stream?
> As far as I understand the fetch action is handled by the _listenLNUrl_ function that returns a stream directly to the _lnurl_handler.dart_.

`_listenLNURL` captures the first request from client to server.  (Step. 2 and Step 3 in the [spec](https://github.com/fiatjaf/lnurl-rfc/blob/master/lnurl-pay.md).

In step 5 the client makes a second request with an amount to get an invoice from the server.

I'm re-using the PayFetchResponse class and updating its amount for step 5 but I should make a new class to make the difference between step 2 and step 5 clear.

Reply:
I see. Makes sense overall.
I will do another review for the new commits later today.